packages/casting/src/follower-cosmjs.js

```diff
@@ -210,71 +201,64 @@ export const makeCosmjsFollower = (
       arrayEqual(result.subarray(0, dataPrefixBytes.length), dataPrefixBytes),
       X`${result} doesn't start with data prefix ${dataPrefixBytes}`,
     );
-    return result.slice(dataPrefixBytes.length);
-  };
-
-  /**
-   * @param {number} [height]
-   * @returns {Promise<Uint8Array>}
-   */
-  const getProvenDataAtHeight = async height => {
-    return retryGetDataAndStripPrefix(
-      async (endpoint, storeName, storeSubkey) => {
-        const queryClient = await provideQueryClient(endpoint);
-        return E(queryClient).queryVerified(storeName, storeSubkey, height);
-      },
-    );
+    // truncate 'value' of response
+    return { ...response, value: result.slice(dataPrefixBytes.length) };
   };
 
   /**
-   * @param {number} height
+   * @param {number} [height] desired height, or the latest height if not set
+   * @param {boolean} [prove=false]
+   * @returns {Promise<QueryAbciResponse>}
    */
-  const getUnprovenDataAtHeight = async height => {
+  const getAvailableData = async (height, prove = false) => {
```

Comment:
I recommend using an options bag here to avoid the [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)